sdk/resourcemanagerhybrid/azure-resourcemanager/src/main/java/com/azure/resourcemanager/AzureResourceManager.java

```diff
@@ -313,6 +324,59 @@ public AzureResourceManager withDefaultSubscription() {
         }
     }
 
+    /**
+     * Gets AzureEnvironment from ARM endpoint.
+     *
+     * @param armEndpoint the URL of ARM endpoint.
+     * @param httpClient the HTTP client for requests.
+     * @return the AzureEnvironment
+     * @throws HttpResponseException thrown if the request is rejected by server, or failed to parse metadata.
+     * @throws RuntimeException thrown if ARM endpoint is invalid, or other errors.
+     */
+    public static AzureEnvironment getAzureEnvironmentFromArmEndpoint(String armEndpoint, HttpClient httpClient) {
+        HttpRequest request = new HttpRequest(HttpMethod.GET,
+            String.format("%s/metadata/endpoints?api-version=1.0", armEndpoint))
+            .setHeader("accept", "application/json");
```

Comment:
I've changed this from "application/xml" in sample to "application/json". I assume we wanted JSON instead of XML?

Reply:
Yep, it should be JSON. I guess the endpoint wasn't reading the Accept header before.